torch/fx/proxy.py

```diff
@@ -134,8 +134,13 @@ def iter(self, obj: 'Proxy') -> Iterator:
         we don't know the value of the proxy, but a custom tracer can attach more
         information to the graph node using create_node and can choose to return an iterator.
         """
-        raise TraceError('Proxy object cannot be iterated. '
-                         'This can be attempted when used in a for loop or as a *args or **kwargs function argument.')
+        raise TraceError('Proxy object cannot be iterated. This can be '
+                         'attempted when the Proxy is used in a loop or'
+                         ' as a *args or **kwargs function argument. '
+                         'Consider factoring out the untraceable logic '
+                         'using torch.fx.wrap. See the torch.fx docs on'
+                         ' pytorch.org for a more detailed explanation '
+                         'of what types of control flow can be traced.')
```

Comment:
I think there's another alternative: if the control flow is static (e.g. the loop trip count is based on some hyperparameter), the code can be refactored as something like:

```
for i in range(self.some_hyperparameter):
  indexed_item = proxied_value[i]
```

We might want to work something like that in as well

Reply:
Good idea! I moved some of that information up into the Proxy docstring to give it more visibility. Please check over the new things I added before I land the fixes